framework/bundles/org.jboss.tools.rsp.server/src/main/java/org/jboss/tools/rsp/server/filewatcher/FileWatcherService.java

```diff
@@ -175,19 +182,39 @@ private void subscribeSinglePath(Path working) {
 			// The service should return the same watchkey for 
 			// the same path, assuming the folder hasn't been deleted
 			// and recreated. 
-			WatchKey k = working.register(watchService, StandardWatchEventKinds.ENTRY_CREATE, 
-			        StandardWatchEventKinds.ENTRY_DELETE,  StandardWatchEventKinds.ENTRY_MODIFY);
+			WatchKey key = register(working);
 			WatchKey existing = subscriptions.get(working);
-			if( !k.equals(existing)) {
-				subscriptions.put(working, k);
+			if( !key.equals(existing)) {
+				subscriptions.put(working, key);
 				if( existing != null )
 					existing.cancel();
 			}
 		} catch (IOException e) {
 			log(e);
 		}
 	}
-	
+
+	private WatchKey register(Path working) throws IOException {
+		WatchEvent.Kind<?>[] watchEventKinds = new WatchEvent.Kind<?>[] { 
+			StandardWatchEventKinds.ENTRY_CREATE,
+			StandardWatchEventKinds.ENTRY_DELETE, 
+			StandardWatchEventKinds.ENTRY_MODIFY };
+		HighSensitivityWatchEventModifier highSensivity = new HighSensitivityWatchEventModifier();
+
+		if (highSensivity.isRequired()) {
+			if (!highSensivity.exists()) {
+				LOG.warn("Watchservice requires high sensitivity on this system."
```

Comment:
In this patch we're logging a warning if we see high sensitivity being necessary but not available. Maybe we should provide ways for the UI to present a dialog to the user in this case.
@robstryker @jeffmaury thoughts?

Reply:
The only issue is that the message will be logged only when the service will be used